### PR TITLE
Update multus deployment, configuration, and verification

### PR DESCRIPTION
**Pull request by clacroix12**

### CODE REVIEW

**Comment by clacroix12** [comment]
I've deployed successfully as well as ran the deployment verification successfully against this cluster.

Deployment: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/7729/
Verification: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/23853/

This is working for one scenario in which we configure both the public and cluster networks.

**Comment by clacroix12** [comment]
There are some additional steps needed to complete verification, so moving this back to WIP

**Review by OdedViner — DISMISSED** [review]

File: conf/deployment/vsphere/upi_1az_rhcos_multus_vsan_3m_3w.yaml [hunk 1]
```diff
@@ -13,6 +13,8 @@ ENV_DATA:
   compute_memory: '65536'
   fio_storageutilization_min_mbps: 10.0
   is_multus_enabled: true
+  multus_create_public_net: true
+  multus_create_cluster_net: true
```
Comment (ebenahar) [inline conf/deployment/vsphere/upi_1az_rhcos_multus_vsan_3m_3w.yaml:17]: is the plan to create configuration files for covering more permutations of public net and cluster net?

File: ocs_ci/deployment/deployment.py [hunk 2]
```diff
@@ -652,22 +652,74 @@ def deploy_ocs_via_operator(self, image=None):
         logger.info("Creating namespace and operator group.")
         run_cmd(f"oc create -f {constants.OLM_YAML}")
 
-        # create multus network
+        # Create Multus Networks
```
Comment (ebenahar) [inline ocs_ci/deployment/deployment.py:655]: are we doing the same also for nodes added post deployment?

File: conf/deployment/vsphere/upi_1az_rhcos_multus_vsan_3m_3w.yaml (hunk 1 above)
Comment (clacroix12) [reply conf/deployment/vsphere/upi_1az_rhcos_multus_vsan_3m_3w.yaml:17]: Yes, should have those configs in today.

File: ocs_ci/deployment/deployment.py (hunk 2 above)
Comment (clacroix12) [reply ocs_ci/deployment/deployment.py:655]: I haven't added anything regarding nodes added post-deployment. I suspect we would want to enable PROMISC on the network interface for the newly added node. Are we running add_node tests in vsphere environments?

**Comment by clacroix12** [comment]
Added the ability to create either the public or cluster network individually as well as both (default behavior). I've also added configurations that should cover more of the deployment scenario permutations as described in our Multus deployment testing document: https://url.corp.redhat.com/6d49eae

File: conf/deployment/baremetal/upi_1az_rhcos_multus_cluster_nvme_intel_3m_3w copy.yaml
Comment (OdedViner) [inline conf/deployment/baremetal/upi_1az_rhcos_multus_cluster_nvme_intel_3m_3w copy.yaml:1]: @clacroix12 Can you fix the conf file name
`conf/deployment/baremetal/upi_1az_rhcos_multus_cluster_nvme_intel_3m_3w copy.yaml `->
`conf/deployment/baremetal/upi_1az_rhcos_multus_cluster_nvme_intel_3m_3w.yaml`

Comment (clacroix12) [reply conf/deployment/baremetal/upi_1az_rhcos_multus_cluster_nvme_intel_3m_3w copy.yaml:1]: Fixed

File: ocs_ci/deployment/deployment.py (hunk 2 above)
Comment (ebenahar) [reply ocs_ci/deployment/deployment.py:758]: Yes, add node tests are done on vSphere (I believe both UPI and IPI).
@clacroix12 @OdedViner should we handle the adjustments needed for Multus in add node functionality with this PR or a follow up one?

**Review by petr-balogh — APPROVED** [review]

File: conf/deployment/vsphere/upi_1az_rhcos_multus_cluster_vsan_3m_3w .yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+---
+# This config is suppose to work on most of DCs we have.
```
[the remark this answers is not in the document]
Comment (clacroix12) [reply conf/deployment/vsphere/upi_1az_rhcos_multus_cluster_vsan_3m_3w .yaml]: I had already fixed this, not sure how it was changed back, possibly due to rebasing. This should be fixed now.

**Review by ebenahar — APPROVED** [review]

**Review by OdedViner — APPROVED** [review]

File: ocs_ci/deployment/deployment.py (hunk 2 above)
Comment (clacroix12) [reply ocs_ci/deployment/deployment.py:788]: @ebenahar missed this update earlier but yes I think any changes necessary for test cases to work with multus deployments should be done in isolated PRs.

**Review by petr-balogh — APPROVED** [review]

**Pull request merged**